### ISSUE TITLE
Tree shaking 

### ISSUE THREAD

**Issue by rubydesign** [pissang/claygl#58]
Hi
this looks like a great project, thanks for sharing.

It doesn't look to big for starters, but i was curious how to achieve the 40k zipped.
I'm not a js buff (don't know the tooling, like rollup), maybe you could share a simple example.

I have been coding for a while, if there is any small boring tasks you want to offload, just say.

**Comment by pissang** [comment]
Thanks! The core rendering modules of ClayGL is very small.  Here a simple gist which bundle file is only 39kb after gzipped.

https://gist.github.com/pissang/7e04ced163da9f727a2a6e5244547393

This minimal project includes basic scene management(stuff like scene tree, material, geometry, camera), rendering management and animations. Besides these, there are no built-in shaders, post processing, particle system, model loader. Which are also used frequently in the 3d programming.

It' useful when you want to build another library upon ClayGL and want to keep the size as small as possible. Like https://github.com/ecomfe/echarts-gl .

It will be great if you can contribute some examples or tutorials when using the ClayGL. Suggestions or bug reports are also welcomed. Thanks a lot!

**Comment by rubydesign** [comment]
Very nice, thank you.
Didn't have a clue it would be so simple (the webpacker config i mean).
Btw, would suggest putting a little build and usage instructions in the readme, and i could make a wiki page from this if you want

**Comment by rubydesign** [comment]
@pissang I got the shaking to work. Though the example you gave needed some tweeking. Had to learn some webpack and add a resolve to get the directories/names to match. I had about 50k, but that's good enough. 
I then tried to use my simple example, but that used the application and i got a bigger pack than your dist (but webpack does put a lot of crap in there as comments). 

Anyway, this encourages me to change my app to clay, and like i said i can help a bit in the process.

I will try to avoid the application (for better shaking) and while i like the example in the readme, it is a little more difficult to find what exactly the process is to do it oneself. I see a lot of the examples are still written without application, so i am guessing that was introduced later. 
It looks like some code could be pushed down from there, i'll open issues.
